### DISCUSSION TITLE
Integrating within existing spring boot config

### DISCUSSION THREAD

**Discussion by ahsanbagwan** [yahoo/elide#1797] [Elide 4]
This is how I retrofitted Search Store within existing JPA data store. Does this look correct?

```
 @Bean
  public Elide elide(SpringHibernateDataStore springHibernateDataStore, ObjectMapper objectMapper, EntityDictionary entityDictionary, ExtendedAuditLogger extendedAuditLogger) {
    RSQLFilterDialect rsqlFilterDialect = new RSQLFilterDialect(entityDictionary, new CaseSensitivityStrategy.UseColumnCollation());

    registerAdditionalConverters();

    // Wrap store in search store
    EntityManagerFactory emf = Persistence.createEntityManagerFactory("searchDataStore");
    boolean indexOnStartup = true;
    DataStore searchStore = new SearchDataStore(springHibernateDataStore, emf, indexOnStartup);

    return new Elide(new ElideSettingsBuilder(searchStore)
      .withJsonApiMapper(new JsonApiMapper(entityDictionary, objectMapper))
      .withAuditLogger(extendedAuditLogger)
      .withEntityDictionary(entityDictionary)
      .withJoinFilterDialect(rsqlFilterDialect)
      .withSubqueryFilterDialect(rsqlFilterDialect)
      .build());
  }

  @Bean
  SpringHibernateDataStore springHibernateDataStore(PlatformTransactionManager txManager,
                                                    AutowireCapableBeanFactory beanFactory,
                                                    EntityManager entityManager) {
    return new SpringHibernateDataStore(txManager, beanFactory, entityManager, false, true, ScrollMode.FORWARD_ONLY);
  }
```

While running the application I also face this error. Why would it say it cannot find the `definition`?

`Caused by: org.hibernate.search.exception.SearchException: Lucene analyzer found with an unknown definition: case_insensitive`

**Comment by aklish** [comment]
I would construct both the JPA and Search stores in this bean:

https://github.com/yahoo/elide/blob/elide-4.x/elide-spring/elide-spring-boot-autoconfigure/src/main/java/com/yahoo/elide/spring/config/ElideAutoConfiguration.java#L115-L120

Here is an example of the SearchDataStore being initialized:
https://github.com/yahoo/elide/blob/elide-4.x/elide-datastore/elide-datastore-search/src/test/java/com/yahoo/elide/datastores/search/DependencyBinder.java#L58-L62

However, I don't think the error above is related to the code snippets above.

**Reply by ahsanbagwan** [reply]
Thanks for your response.

I'm not sure how two store could be constructed within this bean as per this link posted above

https://github.com/yahoo/elide/blob/elide-4.x/elide-spring/elide-spring-boot-autoconfigure/src/main/java/com/yahoo/elide/spring/config/ElideAutoConfiguration.java#L115-L120

**Reply by aklish** [reply]
In this case, the Search store wraps and delegates to the JPA Store.  The bean should return the Search store (which wraps the JPA store).
